pkg/controller/datadogagent/clusteragent_metricsserver.go

```diff
@@ -43,34 +41,95 @@ func buildMetricsServerClusterRoleBinding(dda *datadoghqv1alpha1.DatadogAgent, n
 	return nil
 }
 
-func (r *ReconcileDatadogAgent) deleteIfNeededHpaClusterRoleBinding(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, name, agentVersion string, clusterRoleBinding *rbacv1.ClusterRoleBinding) (reconcile.Result, error) {
-	newClusterRoleBinding := buildMetricsServerClusterRoleBinding(dda, name, agentVersion)
-	if newClusterRoleBinding != nil && clusterRoleBinding != nil && !apiequality.Semantic.DeepEqual(&rbacv1.ClusterRoleBinding{}, clusterRoleBinding.Subjects) {
-		// External Metrics Server used for HPA has been disabled
-		// Delete its ClusterRoleBinding
-		logger.V(1).Info("deleteClusterAgentHPARoleBinding", "clusterRoleBinding.name", clusterRoleBinding.Name)
-		event := buildEventInfo(clusterRoleBinding.Name, clusterRoleBinding.Namespace, clusterRoleBindingKind, datadog.DeletionEvent)
-		r.recordEvent(dda, event)
-		if err := r.client.Delete(context.TODO(), clusterRoleBinding); err != nil {
-			if errors.IsNotFound(err) {
-				return reconcile.Result{}, nil
-			}
-			return reconcile.Result{}, err
+func (r *ReconcileDatadogAgent) createHPAClusterRoleBinding(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, name, agentVersion string) (reconcile.Result, error) {
+	clusterRoleBinding := buildMetricsServerClusterRoleBinding(dda, name, agentVersion)
+	if clusterRoleBinding == nil {
+		return reconcile.Result{}, nil
```

Comment:
Could we log something here ?

Reply:
Definitely though what concern are we would be addressing here?

Reply:
It's just a nit, but if we fail to build the CRB, there is no error surfaced https://github.com/DataDog/datadog-operator/pull/137/files#diff-bbe5868bca18c548b30bed831ffd53acR20-R42 so a log could help imo

Reply:
So the way it can be nil is if `!isMetricsProviderEnabled`, so we are just guarding trying to create it by that. This is not a new func btw. The diff is a bit off on this file: https://github.com/DataDog/datadog-operator/pull/137/files#diff-bbe5868bca18c548b30bed831ffd53acL67